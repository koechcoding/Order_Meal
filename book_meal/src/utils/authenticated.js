export default function authenticated(){
    try{
        return JSON.parse(localStorage.getItem('user'));
    } catch(err){
        
    }
}