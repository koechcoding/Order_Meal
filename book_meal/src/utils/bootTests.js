import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter()});

global.window = {}
import localStorage from 'mock-local-Storage';
window.localStorage = global.localStorage

export default Enzyme;
