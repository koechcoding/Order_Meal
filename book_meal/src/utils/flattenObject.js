export default function flattenObject(obj){
    let result = {}
    for (const key in obj){
        if((typeof obj[key])=== 'object' &&  obj[key]  !== null) {
            let flattened = flattenObject(obj[key])
            for(const key2 in flattened){
                result[key + "." + key2] = flattened[key2];
            }
        }else{
            result[key] = obj[key];
        }
    }
    return result;
}