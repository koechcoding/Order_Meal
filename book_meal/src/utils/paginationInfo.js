export default function paginationInfo(page){

    if(!page){
        return{}
    }

    let currentPage = 1
    if(page.next_page)
         currentPage = page.next_page -1;
    else if (page.prev_page)
        currentPage = page.prev_page + 1;

    let pageCount = Math.ceil(page.total / page.per_paage);
    if(Number.isNaN(pageCount) || pageCount === 0)
      pageCount = 1; 

    return {
        pageCount,
        currentPage,
        total: page.total,
        perPage: page.per_page,
        hasNext: page.has_next,
        hasPrev: page.has_prev,
        nextPage: page.next_page,
        prevPage: page.prev_page,
        currentCount: page.current_count,
    };
}