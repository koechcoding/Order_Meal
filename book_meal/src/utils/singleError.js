const singleError = (errorPayload)=>{
    if(errorPayload && errorPayload.data){
        const response = errorPayload.data;
        //alternative to Object.keys...
        for(let errorKey in response.errors){
            return response.error[errorKey]
        }
        if(response.message){
            return response.message;
        }
    }
    return 'Unknown error. Please try again!';
}
export default singleError;