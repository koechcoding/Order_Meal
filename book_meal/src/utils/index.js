import singleError from './singleError';
import paginationInfo from './paginationInfo';
import authenticated from './authenticated';
import flattenObject from './flattenObject';

export { 
    singleError, 
    paginationInfo, 
    authenticated,
    flattenObject
}