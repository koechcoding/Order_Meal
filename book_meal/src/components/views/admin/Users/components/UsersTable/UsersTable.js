import React from 'react';
import PropTypes from 'prop-types';
import Paginator from 'src/components/common/Paginator';
import { flattenObject, paginationInfo } from 'src/utils';

import Table from 'src/components/common/Table';
import { EntryType } from 'src/constants';

class UsersTable extends React.Component {

    render() {
        const { users = []  } = this.props.data;
        let rows = users.map(row => flattenObject(row));
        const tableData = {
            columns: [
                { key: 'id', title: 'ID', type: EntryType.NUMBER },
                { key: 'email', title: 'Email', type: EntryType.TEXT },
                { key: 'username', title: 'Name', type: EntryType.TEXT },
                { key: 'created_at', title: 'Created On', type: EntryType.DATE },
                { key: 'role', title: 'Role', type: EntryType.ROLE }
            ],
            rows,
        };
        const { onToggle } = this.props;
        const pageInfo = paginationInfo(this.props.data);
        return (
            <div>
                <Table onToggle={onToggle} data={tableData} />
                <Paginator {...this.props} pageInfo={pageInfo} />
            </div>
        );
    }
}

UsersTable.propTypes = {
    data: PropTypes.object
}

export default UsersTable;