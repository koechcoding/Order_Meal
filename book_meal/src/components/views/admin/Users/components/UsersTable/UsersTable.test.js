import React from 'react';
import { shallow }  from 'enzyme';
import 'src/utils/bootTests'; 
import UsersTable from './UsersTable';
import Adapter from 'enzyme-adapter-react-16';
const props = {
    data: {
        users: [{
            id: 1,
            email: 'mail@test.com',
            username: 'edward',
            created_at: '10-08-2018',
            role: 1,
        }]
    },
    onDelete: () => {},
    onEdit: () => {},
    onNext: () => {},
    onPrev: () => {},
    onToggle: () => {},
}

describe('<UsersTable />', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<UsersTable {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});