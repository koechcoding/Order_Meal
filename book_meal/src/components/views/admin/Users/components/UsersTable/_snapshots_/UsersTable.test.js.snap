// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersTable /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UsersTable
    data={
        Object {
            "users": Array [
              Object {
                "created_at": "10-08-2018",
                "email": "mail@test.com",
                "id": 1,
                "role": 1,
                "username": "edward",
              },
            ],
          }
    }
    onDelete={[Function]}
    onEdit={[Function]}
    onNext={[Function]}
    onPrev={[Function]}
    onToggle={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Table
          data={
                    Object {
                              "columns": Array [
                                Object {
                                  "key": "id",
                                  "title": "ID",
                                  "type": 1,
                                },
                                Object {
                                  "key": "email",
                                  "title": "Email",
                                  "type": 0,
                                },
                                Object {
                                  "key": "username",
                                  "title": "Name",
                                  "type": 0,
                                },
                                Object {
                                  "key": "created_at",
                                  "title": "Created On",
                                  "type": 3,
                                },
                                Object {
                                  "key": "role",
                                  "title": "Role",
                                  "type": 5,
                                },
                              ],
                              "rows": Array [
                                Object {
                                  "created_at": "10-08-2018",
                                  "email": "mail@test.com",
                                  "id": 1,
                                  "role": 1,
                                  "username": "edward",
                                },
                              ],
                            }
          }
          onToggle={[Function]}
/>,
        <Paginator
          data={
                    Object {
                              "users": Array [
                                Object {
                                  "created_at": "10-08-2018",
                                  "email": "mail@test.com",
                                  "id": 1,
                                  "role": 1,
                                  "username": "edward",
                                },
                              ],
                            }
          }
          onDelete={[Function]}
          onEdit={[Function]}
          onNext={[Function]}
          onPrev={[Function]}
          onToggle={[Function]}
          pageInfo={
                    Object {
                              "currentCount": undefined,
                              "currentPage": 1,
                              "hasNext": undefined,
                              "hasPrev": undefined,
                              "nextPage": undefined,
                              "pageCount": 1,
                              "perPage": undefined,
                              "prevPage": undefined,
                              "total": undefined,
                            }
          }
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "columns": Array [
              Object {
                "key": "id",
                "title": "ID",
                "type": 1,
              },
              Object {
                "key": "email",
                "title": "Email",
                "type": 0,
              },
              Object {
                "key": "username",
                "title": "Name",
                "type": 0,
              },
              Object {
                "key": "created_at",
                "title": "Created On",
                "type": 3,
              },
              Object {
                "key": "role",
                "title": "Role",
                "type": 5,
              },
            ],
            "rows": Array [
              Object {
                "created_at": "10-08-2018",
                "email": "mail@test.com",
                "id": 1,
                "role": 1,
                "username": "edward",
              },
            ],
          },
          "onToggle": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Object {
            "users": Array [
              Object {
                "created_at": "10-08-2018",
                "email": "mail@test.com",
                "id": 1,
                "role": 1,
                "username": "edward",
              },
            ],
          },
          "onDelete": [Function],
          "onEdit": [Function],
          "onNext": [Function],
          "onPrev": [Function],
          "onToggle": [Function],
          "pageInfo": Object {
            "currentCount": undefined,
            "currentPage": 1,
            "hasNext": undefined,
            "hasPrev": undefined,
            "nextPage": undefined,
            "pageCount": 1,
            "perPage": undefined,
            "prevPage": undefined,
            "total": undefined,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Table
            data={
                        Object {
                                    "columns": Array [
                                      Object {
                                        "key": "id",
                                        "title": "ID",
                                        "type": 1,
                                      },
                                      Object {
                                        "key": "email",
                                        "title": "Email",
                                        "type": 0,
                                      },
                                      Object {
                                        "key": "username",
                                        "title": "Name",
                                        "type": 0,
                                      },
                                      Object {
                                        "key": "created_at",
                                        "title": "Created On",
                                        "type": 3,
                                      },
                                      Object {
                                        "key": "role",
                                        "title": "Role",
                                        "type": 5,
                                      },
                                    ],
                                    "rows": Array [
                                      Object {
                                        "created_at": "10-08-2018",
                                        "email": "mail@test.com",
                                        "id": 1,
                                        "role": 1,
                                        "username": "edward",
                                      },
                                    ],
                                  }
            }
            onToggle={[Function]}
/>,
          <Paginator
            data={
                        Object {
                                    "users": Array [
                                      Object {
                                        "created_at": "10-08-2018",
                                        "email": "mail@test.com",
                                        "id": 1,
                                        "role": 1,
                                        "username": "edward",
                                      },
                                    ],
                                  }
            }
            onDelete={[Function]}
            onEdit={[Function]}
            onNext={[Function]}
            onPrev={[Function]}
            onToggle={[Function]}
            pageInfo={
                        Object {
                                    "currentCount": undefined,
                                    "currentPage": 1,
                                    "hasNext": undefined,
                                    "hasPrev": undefined,
                                    "nextPage": undefined,
                                    "pageCount": 1,
                                    "perPage": undefined,
                                    "prevPage": undefined,
                                    "total": undefined,
                                  }
            }
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "columns": Array [
                Object {
                  "key": "id",
                  "title": "ID",
                  "type": 1,
                },
                Object {
                  "key": "email",
                  "title": "Email",
                  "type": 0,
                },
                Object {
                  "key": "username",
                  "title": "Name",
                  "type": 0,
                },
                Object {
                  "key": "created_at",
                  "title": "Created On",
                  "type": 3,
                },
                Object {
                  "key": "role",
                  "title": "Role",
                  "type": 5,
                },
              ],
              "rows": Array [
                Object {
                  "created_at": "10-08-2018",
                  "email": "mail@test.com",
                  "id": 1,
                  "role": 1,
                  "username": "edward",
                },
              ],
            },
            "onToggle": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Object {
              "users": Array [
                Object {
                  "created_at": "10-08-2018",
                  "email": "mail@test.com",
                  "id": 1,
                  "role": 1,
                  "username": "edward",
                },
              ],
            },
            "onDelete": [Function],
            "onEdit": [Function],
            "onNext": [Function],
            "onPrev": [Function],
            "onToggle": [Function],
            "pageInfo": Object {
              "currentCount": undefined,
              "currentPage": 1,
              "hasNext": undefined,
              "hasPrev": undefined,
              "nextPage": undefined,
              "pageCount": 1,
              "perPage": undefined,
              "prevPage": undefined,
              "total": undefined,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;